Source/bmalloc/bmalloc/TZoneHeap.h

```diff
@@ -194,7 +194,7 @@ private: \
 private: \
     static _exportMacro BNO_INLINE void* operatorNewSlow(size_t size) \
     { \
-        static const TZoneSpecification s_heapSpec = { &s_heapRef, sizeof(_type), CompactAllocationMode:: _compactMode, SizeAndAlignment::encode<_type>() TZONE_SPEC_NAME_ARG(#_type) }; \
+        static const TZoneSpecification s_heapSpec = { &s_heapRef, sizeof(_type), ::bmalloc::api::requiresCompactPointers<_type>() ? CompactAllocationMode::Compact : CompactAllocationMode:: _compactMode, SizeAndAlignment::encode<_type>() TZONE_SPEC_NAME_ARG(#_type) }; \
```

Comment:
Would it be possible to create a `constexpr` template function `compactModeForType` that implements this `?:` logic by taking a `CompactAllocationMode` argument.  It can be a peer of `requiresCompactPointers`.  Seems that  would be better than having this logic in 3 places.

Reply:
Seems reasonable, I'll give it a shot.